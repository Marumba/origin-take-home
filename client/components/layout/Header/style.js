import styled from '@emotion/styled';
import { css } from '@emotion/react';

export const Header = styled.header`
	background: #fff;

	display: flex;
	justify-content: left;
	align-items: center;

	padding: 12px;

	transition: padding ${({ theme }) => theme.animations.durations.fastest} ease-in;

	@media ${({ theme }) => theme.medias.desktop} {
		padding: 24px;
	}
`;

export const originLogo = theme => css`
	height: 24px;

	transition: height ${theme.animations.durations.fastest} ease-in;

	@media ${theme.medias.desktop} {
		height: 32px;
	}
`;
