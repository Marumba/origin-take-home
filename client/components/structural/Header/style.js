import styled from '@emotion/styled';
import { css } from '@emotion/react';

const transitionTime = '200ms';

export const Header = styled.header`
	background: #fff;

	display: flex;
	justify-content: left;
	align-items: center;

	padding: 12px;

	transition: padding ${transitionTime} ease-in;

	@media ${({ theme }) => theme.medias.desktop} {
		padding: 24px;
	}
`;

export const originLogo = theme => css`
	height: 24px;

	transition: height ${transitionTime} ease-in;

	@media ${theme.medias.desktop} {
		height: 32px;
	}
`;
