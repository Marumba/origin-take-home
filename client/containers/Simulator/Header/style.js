import styled from '@emotion/styled';

export const Header = styled.div`
	display: flex;
	align-items: center;
	justtify-content: left;

	svg {
		margin-right: 15px;
	}

	${({ customCss, theme }) => (typeof customCss === 'function' ? customCss(theme) : customCss)}
`;

export const TextWrapper = styled.div`
	display: flex;
	align-items: left;
	justify-content: center;
	flex-direction: column;
`;
export const Title = styled.h2`
	${({ theme }) => theme.mixins.headingSmall};

	display: block;
`;
export const Subtitle = styled.h3`
	display: block;
`;
