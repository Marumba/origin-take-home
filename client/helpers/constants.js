export const CURRENCY_CONFIG = {
	USD: {
		locale: 'en-US',
		options: {
			style: 'currency',
			currency: 'USD',
			maximumFractionDigits: 2,
			minimumFractionDigits: 2
		}
	}
};

export const MONTHS = [
	'January',
	'February',
	'March',
	'April',
	'May',
	'June',
	'July',
	'August',
	'September',
	'October',
	'November',
	'December'
];
