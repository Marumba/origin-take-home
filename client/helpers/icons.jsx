import { ReactComponent as USD } from '~/assets/imgs/icons/icon-dollar-sign.svg';
import { ReactComponent as Home } from '~/assets/imgs/icons/icon-home.svg';
import { ReactComponent as ArrowLeft } from '~/assets/imgs/icons/icon-arrow-left.svg';

export const CurrencyIcons = {
	USD
};

export const Icons = { Home, ArrowLeft, ArrowRight: () => <ArrowLeft transform="rotate(180)" /> };
