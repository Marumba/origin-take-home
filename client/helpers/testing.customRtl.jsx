import React from 'react';
import { RecoilRoot } from 'recoil';
import { render as rtlRender } from '@testing-library/react';
import { ThemeProvider } from '@emotion/react';
import { childrenType } from '~/types';

import theme from '~/themes';

function render(ui, { ...renderOptions } = {}) {
	function Wrapper({ children }) {
		return (
			<RecoilRoot>
				<ThemeProvider theme={theme}>{children}</ThemeProvider>
			</RecoilRoot>
		);
	}
	Wrapper.propTypes = {
		children: childrenType.isRequired
	};
	return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from '@testing-library/react';
export { render };
