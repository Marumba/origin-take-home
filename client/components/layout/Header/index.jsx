import React from 'react';

import { OriginLogo } from '~/components/minimal';

import * as S from './style';

function Header() {
	return (
		<S.Header>
			<OriginLogo customCss={S.originLogo} type="withLink" />
		</S.Header>
	);
}

export default Header;
