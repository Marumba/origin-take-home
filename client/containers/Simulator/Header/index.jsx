import React from 'react';
import { string, customCssType, oneOfType, element } from '~/types';

import * as S from './style';

function Header({ customCss, icon, subtitle, title }) {
	return (
		<S.Header data-testid="simulator-header" customCss={customCss}>
			{icon}
			<S.TextWrapper>
				<S.Title className="highEmphasis">{title}</S.Title>
				<S.Subtitle className="lowEmphasis">{subtitle}</S.Subtitle>
			</S.TextWrapper>
		</S.Header>
	);
}

Header.propTypes = {
	customCss: customCssType,
	icon: oneOfType([string, element]),
	subtitle: string,
	title: string
};

Header.defaultProps = {
	customCss: undefined,
	icon: undefined,
	subtitle: '',
	title: ''
};

export default Header;
