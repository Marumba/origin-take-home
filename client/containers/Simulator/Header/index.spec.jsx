import React from 'react';

import { render, screen } from '~/helpers/testing.customRtl';

import { Icons } from '~/helpers/icons';

import Header from '.';

describe('Container Simulator Header', () => {
	it('should render without any props given', () => {
		render(<Header />);
		expect(screen.getByTestId('simulator-header')).toBeInTheDocument();
	});

	it('should display the texts from given props, accordingly', () => {
		const { rerender } = render(<Header title="Title" />);
		expect(screen.getByText('Title')).toBeInTheDocument();

		rerender(<Header subtitle="Subtitle" />);
		expect(screen.getByText('Subtitle')).toBeInTheDocument();

		rerender(<Header subtitle="Subtitle" />);
		expect(screen.getByText('Subtitle')).toBeInTheDocument();

		rerender(<Header icon={<Icons.Home role="img" />} />);
		expect(screen.getByRole('img')).toBeInTheDocument();
	});

	it('should display all', () => {
		const { container } = render(
			<Header subtitle="Subtitle" title="Title" icon={<Icons.Home role="img" />} />
		);
		expect(screen.getByText('Title')).toBeInTheDocument();
		expect(screen.getByText('Subtitle')).toBeInTheDocument();
		expect(screen.getByRole('img')).toBeInTheDocument();

		expect(container.firstChild).toMatchSnapshot();
	});
});
