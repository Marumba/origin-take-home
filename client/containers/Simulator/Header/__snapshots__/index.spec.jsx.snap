// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Simulator Header should display all 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  justtify-content: left;
}

.emotion-0 svg {
  margin-right: 15px;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: left;
  -webkit-box-align: left;
  -ms-flex-align: left;
  align-items: left;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-4 {
  font-family: 'Rubik',sans-serif;
  font-weight: 500;
  font-size: 20px;
  line-height: 120%;
  display: block;
}

@media screen and (min-width: 992px) {
  .emotion-4 {
    font-size: 24px;
  }
}

.emotion-4.highEmphasis {
  color: #1E2A32;
}

.emotion-6 {
  font-family: 'Work Sans',sans-serif;
  font-weight: 400;
  font-size: 14px;
  line-height: 150%;
  display: block;
}

@media screen and (min-width: 992px) {
  .emotion-6 {
    font-size: 16px;
  }
}

.emotion-6.semibold,
.emotion-6 strong {
  font-weight: 600;
}

.emotion-6.highEmphasis {
  color: #1E2A32;
}

.emotion-6.lowEmphasis {
  color: #708797;
}

<div
  class="emotion-0 emotion-1"
  data-testid="simulator-header"
>
  <div
    role="img"
  />
  <div
    class="emotion-2 emotion-3"
  >
    <h2
      class="highEmphasis emotion-4 emotion-5"
    >
      Title
    </h2>
    <h3
      class="lowEmphasis emotion-6 emotion-7"
    >
      Subtitle
    </h3>
  </div>
</div>
`;
